compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1455,16 +1455,29 @@ trait Implicits { self: Typer =>
        *    - If alt2 is preferred over alt1, pick alt2, otherwise return an
        *      ambiguous implicits error.
        */
-      def disambiguate(alt1: SearchResult, alt2: SearchSuccess) = alt1 match {
+      def disambiguate(alt1: SearchResult, alt2: SearchSuccess) = alt1 match
         case alt1: SearchSuccess =>
-          val diff = compareCandidate(alt1, alt2.ref, alt2.level)
+          var diff = compareCandidate(alt1, alt2.ref, alt2.level)
```

Comment:
Maybe make 2 vals instead of a var

Reply:
I think it comes out simpler with a `var`.

Reply:
Ok